src/errors.ts

```diff
@@ -44,10 +44,12 @@ function isConnectionError(err: Error): boolean {
   if (err instanceof SQSError) {
     return (
       err.statusCode === 403 ||
+      err.name === "SQSError" ||
       err.code === "CredentialsError" ||
       err.code === "UnknownEndpoint" ||
       err.code === "AWS.SimpleQueueService.NonExistentQueue" ||
```

Comment:
I think this error is for the v2 SDK. The new error code is `QueueDoesNotExist` as far as I can see from my testing. Is it worth updating this while we're here? The CDK was updated in v4/v5 I think.

Reply:
Hey @alexbaileymembr, if you have a look at the reproduction [here](https://stackblitz.com/github/fraserbenjamin/sqs-consumer-starter) (using SDK v3) you can still see the error gets thrown with the name "SQSError" included. If you have no network connection the error you receive looks like this `{"name":"SQSError","code":"Error","time":"2024-04-12T09:17:41.638Z"}` from which there's not much to go off but still should be handled with a wait to prevent retries.

If you're talking about `AWS.SimpleQueueService.NonExistentQueue` that's not something I've changed but was already in the code and I suspect should remain in the code until 8th September 2024 based on this issue here when v2 will no longer be reported. https://github.com/bbc/sqs-consumer/issues/465

If you'd like to setup a call to run though it and I can make any extra changes you think are necessary that would be fantastic thank you.

Reply:
I see. `SQSError` should handle it in that case. 

The v2 SDK was removed from this consumer in v6.0.0 by the looks of it (https://github.com/bbc/sqs-consumer/releases/tag/v6.0.0) so I'd assume that `AWS.SimpleQueueService.NonExistentQueue` is safe to remove. However, it's not directly related to this issue so I don't know if it's preferred to be handled as a separate issue / task.

Reply:
Yeah thanks @alexbaileymembr, is it now possible to get this merged please? Happy to create a 2nd PR to remove that or remove it in this one if you'd like, just let me know.

Reply:
Sorry mate. Can't help you get it merged. Not on the reviewers list.